common/utils.js

```diff
@@ -1751,9 +1753,17 @@
                 }).catch(function (exception) {
                     // error/login modal was closed
                     if (typeof exception == 'string') return;
+                    var ermrestError = ERMrest.responseToError(exception);
+                    var skipLoginDialog = false;
+
+                    if (ermrestError instanceof ERMrest.UnauthorizedError) {
```

Comment:
It seems like instead of using `responseToError` here, I should check what "code" is returned and if it's `401` I should call `new ERMrest.UnauthorizedError(messageMap.unauthorizedErrorCode, messageMap.unauthenticatedAssetRetrieval)`. 

That's what we do in `recordedit.app.js` at least, or rather used to do before we stated we should use `_responseToError`. I guess in that case though, it is returning an `ermrestJS` error and we are sending it back to `ermrestJS` to rewrap it instead of attaching the properties like I did..

Reply:
Instead of changing the existing attributes in the `ERMrest.UnauthorizedError` in the catch block, you can create a chaise specific error class. Call it the same as `UnauthorizedAssetAccess` (or any other name. I used this name because it's 401 and asset related so "unauthorized access" + "asset"). Then you don't need to add the `skipLoginDialog` and chaise can just show that newly added error class.

```javascript
if (ermrestError instanceof ERMrest. UnauthorizedError) {
  ermrestError = new Errors.UnauthorizedAssetAccess();
}

ErrorService.handleException(ermrestError, true);
```

About authenticate vs. authorize:
I know based on our discussion unauthenticated makes more sense. But the error in http is called "unauthorized access" whether we like it or not. And that's the name that we've used for the error class. So I think we should stick with it and call it the same way (`UnauthorizedAssetAccess`). It will be more confusing as you might think that unauthorized is different from unauthenticated (which in this context, in our code, they are the same).

Reply:
I like this suggestion, and I think it makes sense to handle it this way because it is a `chaise` error, `ermrestJS` wouldn't be trying to do something special for it. And I like that it means not adding another flag to the `handleException` function.